thunder/dynamo/report.py

```diff
@@ -607,6 +615,11 @@ def __init__(self, bsym: BoundSymbol, name: str):
     def __repr__(self):
         return f"<ThunderFusionReport of bound symbol\n{self.nvfusion_bsym}>"
 
+    def benchmark(self, fn: Callable, timer_fn: Callable):
+        compiled_fn = fn(self.nvfusion_bsym)
+        inputs = self.get_inputs()
+        return timer_fn("compiled_fn(*inputs)", globals={"compiled_fn": compiled_fn, "inputs": inputs})
+
     def write_nvfuser_repro(self, folder, file_name=None):
```

Comment:
It would be really nice if these methods could be inherited from a common base, as the comment above mentions. Maybe `FXGraphReport` is that common base, or maybe they need a new base class, like `FXGraphSegment`, and then the `FXGraphReport` and `ThunderFusionReport` classes can define methods that customize how their reproductions are created?